File: src/main/java/fr/vlegall/sochief/service/recipe/IRecipeService.java
package fr.vlegall.sochief.service.recipe;

import fr.vlegall.sochief.contracts.request.RecipeUpsertRequestDto;
import fr.vlegall.sochief.contracts.response.RecipeDetailDto;
import fr.vlegall.sochief.contracts.response.RecipeListItemDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

public interface IRecipeService {
    Page<RecipeListItemDto> search(String q, Long categoryId, Pageable pageable);

    RecipeDetailDto getById(Long id, Integer portions);

    RecipeDetailDto create(RecipeUpsertRequestDto dto, MultipartFile image);

    RecipeDetailDto update(Long id, RecipeUpsertRequestDto dto, MultipartFile image);

    void delete(Long id);
}

File: src/main/kotlin/fr/vlegall/sochief/service/recipe/IRecipeService.kt
package fr.vlegall.sochief.service.recipe

import fr.vlegall.sochief.contracts.request.RecipeUpsertRequestDto
import fr.vlegall.sochief.contracts.response.RecipeDetailDto
import fr.vlegall.sochief.contracts.response.RecipeListItemDto
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface IRecipeService {
    fun search(q: String?, categoryId: Long?, pageable: Pageable): Page<RecipeListItemDto>
    fun getById(id: Long, portions: Int? = null): RecipeDetailDto
    fun create(dto: RecipeUpsertRequestDto): RecipeDetailDto
    fun update(id: Long, dto: RecipeUpsertRequestDto): RecipeDetailDto
    fun delete(id: Long)
}